File: lightbulb-recycleradapters/src/main/java/com/github/RooneyAndShadows/lightbulb/recycleradapters/EasyRecyclerAdapterUpdateCallback.java
package com.github.rooneyandshadows.lightbulb.recycleradapters;

import androidx.recyclerview.widget.ListUpdateCallback;
import androidx.recyclerview.widget.RecyclerView;

class EasyRecyclerAdapterUpdateCallback implements ListUpdateCallback {
        private final RecyclerView.Adapter adapter;
        private final int offset;

        public EasyRecyclerAdapterUpdateCallback(RecyclerView.Adapter adapter, int offset) {
            this.adapter = adapter;
            this.offset = offset;
        }

        @Override
        public void onInserted(int position, int count) {
            adapter.notifyItemRangeInserted(position + offset, count);
        }

        @Override
        public void onRemoved(int position, int count) {
            adapter.notifyItemRangeRemoved(position + offset, count);
        }

        @Override
        public void onMoved(int fromPosition, int toPosition) {
            adapter.notifyItemMoved(fromPosition + offset, toPosition);
        }

        @Override
        public void onChanged(int position, int count, Object payload) {
            adapter.notifyItemRangeChanged(position + offset, count, payload);
        }
    }

File: lightbulb-recycleradapters/src/main/java/com/github/RooneyAndShadows/lightbulb/recycleradapters/abstraction/EasyRecyclerAdapterUpdateCallback.kt
package com.github.rooneyandshadows.lightbulb.recycleradapters.abstraction

import androidx.recyclerview.widget.ListUpdateCallback
import androidx.recyclerview.widget.RecyclerView

internal class EasyRecyclerAdapterUpdateCallback(
    private val adapter: RecyclerView.Adapter<*>,
    private val offset: Int
) : ListUpdateCallback {
    override fun onInserted(position: Int, count: Int) {
        adapter.notifyItemRangeInserted(position + offset, count)
    }

    override fun onRemoved(position: Int, count: Int) {
        adapter.notifyItemRangeRemoved(position + offset, count)
    }

    override fun onMoved(fromPosition: Int, toPosition: Int) {
        adapter.notifyItemMoved(fromPosition + offset, toPosition)
    }

    override fun onChanged(position: Int, count: Int, payload: Any?) {
        adapter.notifyItemRangeChanged(position + offset, count)
    }
}